serpentTools/parsers/results.py

```diff
@@ -1,15 +1,247 @@
+        if currVar is not None:
+            # return serpent-style and python-style names
+            return currVar.group(), convertVariableName(currVar.group())
+
+    def _getVarValues(self, tline):
+        """Obtains the values of any variable."""
+        str, vec, scl = self._regexpStr.search(tline),\
+                        self._regexpVec.search(tline),\
+                        self._regexpScl.search(tline)
+        varType, varVals = [], []
+        if str is not None:
+            varType = 'string'
+            varVals = tline[str.span()[0]+1:str.span()[1]-1]
+        elif vec is not None:
+            varType = 'vector'
+            varVals = tline[vec.span()[0]+1:vec.span()[1]-2]
+        elif scl is not None:
+            varType = 'scalar'
+            varVals = tline[scl.span()[0]+1:scl.span()[1]-2]
+        return varType, varVals
+
+    def _precheck(self):
+        """do a quick scan to ensure this looks like a results file."""
+        #MapStrVersions = self._mapSerpentVersion()
+        if self.__serpentVersion in MapStrVersions:
+            self._keysVersion = MapStrVersions[self.__serpentVersion]
+        else:
+            warning("Version {} is not supported by the "
+                                        "ResultsReader".format(self.__serpentVersion))
+        with open(self.filePath) as fid:
+            if fid is None:
+                raise IOError("Attempting to read on a closed file.\n"
+                              "Parser: {}\nFile: {}".format(self, self.filePath))
+            for tline in fid:
+                if self._keysVersion['meta'] in tline:
+                    varType, varVals = self._getVarValues(tline)  # actual serpent version used
+                    if self.__serpentVersion not in varVals:
+                        warning("Version {} is used, but version {} is defined in settings"
+                                                    .format(varVals, self.__serpentVersion))
+                if self._keysVersion['univ'] in tline:
+                    varType, varVals = self._getVarValues(tline)  # universe
+                    self._univlist.append(varVals)
+                    nUniv = self._getNumUniv()
+                    if len(self._univlist) > nUniv:
+                        self._counter['univ'] = nUniv
+                        self._univlist = []
+                        return
+            if not self._univlist:
+                raise SerpentToolsException("No universes are found in the "
+                                            "file {}".format(self.filePath))
+            else:
+                self._counter['univ'] = 1
+                self._univlist = []
+
+    def _postcheck(self):
+        """ensure the parser grabbed expected materials."""
+        if not self.resdata and not self.metadata and not self.univdata:
+            raise SerpentToolsException("No results were collected "
+                                        "from {}".format(self.filePath))
```

Comment:
Initially:
self.universes is set to {}
If no universes are found then technically there will be no branching states and thus self.universes will remain {}.
I do agree that the data within each universe can be empty, but at least we know that branching states were created.

Reply:
Then the only way this would be False would be if there was no universe data in the file (and everything else was empty), correct?

What I'm trying to get at is we already scan for universes, and if none are found, the the _precheck method fails and we never get to the _postcheck method. The intent here is to inform the user that we didn't store any data, results, metadata, or universes.  The break down of the missing data could be

1. No results - bad file or bad user settings
1. No metadata - bad file or bad user settings
1. No universes - bad file

The exception being presented here does not, to me, reflect what is raising the error. As a user, do I need to change some of my settings, or is the file bad? We *should* know if the file is bad if it doesn't contain universes via the _precheck method.